src/renderers/WebGLRenderer.js

```diff
@@ -1536,6 +1536,10 @@ function WebGLRenderer( parameters ) {
 
 		if ( programChange ) {
 
+			parameters.uniforms = programCache.getUniforms( material, parameters );
+
+			material.onBeforeCompile( parameters, _this );
```

Comment:
Need to pass in `parameters` so user code actually modifies the correct reference to `vertexShader` and `fragmentShader`.